CHANGELOG.md

```diff
@@ -29,6 +30,17 @@ librdkafka v1.6.0 is feature release:
    supported on Apache Kafka 2.5 or later, on earlier releases you will
    need to use one producer per input partition for EOS. This limitation
    is not enforced by the producer or broker.
+ * Error handling for the transactional producer has been improved, see
+   the **Transactional Producer fixes** below for more information.
+
+
+## Known issues
+
+ * The Transactional Producer's API timeout handling is inconsistent with the
+   underlying protocol requests, it is therefore strongly recommended that
+   applications call `rd_kafka_commit_transaction()` and
+   `rd_kafka_abort_transaction()` with the `timeout_ms` parameter
+   set to `-1`, which will use the remaining transaction timeout.
```

Comment:
when would someone ever want a different timeout?
is there are reason to give people a way to shoot themselves in the foot?
i guess backwards-compatibility?
maybe there is an upgrade path whereby in 1.6 you get a warning if you provide a timeout != -1 and then in 2.0 you break it.

Reply:
These calls are supposed to be continuable on timeout so that an app that does not want to block for too long can call it repeatedly in a loop, that is not currently the case though and the transaction is set to an abortable state on timeout. We should fix the retriable/continuable thing which will also allow any timeout.